seqr/views/utils/dataset_utils.py

```diff
@@ -99,64 +99,70 @@ def _get_individual_sample_lookup(individuals):
     return {i.individual_id: i for i in individuals}
 
 
-def _get_mapped_individual_lookup_key(sample_id_to_individual_id_mapping):
+def _get_mapped_individual_lookup_key(sample_id_to_individual_id_mapping, is_rna=False):
     sample_id_to_individual_id_mapping = sample_id_to_individual_id_mapping or {}
 
-    def _get_mapped_id(sample_id):
-        return sample_id_to_individual_id_mapping.get(sample_id, sample_id)
-    return _get_mapped_id
+    def _get_mapped_key(sample_key):
+        if is_rna:
+            return sample_id_to_individual_id_mapping.get(sample_key[0], sample_key[0]), sample_key[1]
+        return sample_id_to_individual_id_mapping.get(sample_key, sample_key)
+    return _get_mapped_key
+
+
+def _get_sample_id_from_dict_key(key):
+    return key[0] if isinstance(key, tuple) else key
 
 
 def _find_or_create_missing_sample_records(
         samples,
         projects,
         user,
-        sample_ids,
+        sample_count,
```

Comment:
The `sample_ids` is replaced by the `sample_count` because we have moved the `remaining_sample_keys` up to the caller function.